### PR TITLE
[NEEDS TM] [PORT] Ports Pixelshifting

### PR DESCRIPTION
**Pull request by fighterslam**
## About The Pull Request
Ports Pixelshifting from https://github.com/BeeStation/BeeStation-Hornet/pull/6060.

> PIXEL SHIFTING:
Adds pixel shifting into the game, enabling you to smoothly shift around on the tile you're in. With a few changes.
Movement is now smooth, you do not need to spam the key you wish to move in, you may hold it down and watch yourself smoothly move.
Keybind is currently C.

_This PR does not contain a savefile Burn, existing users will not have this key bound by default. A burn will be added when it is fullmerge ready._

BASEPIXELS
> This adds two variables (base_pixel_x and base_pixel_y) and refactors a little of the pixel-shifting code to support better persistent pixel-shifts that would otherwise get reset by most animation and standard shifting procs.
> 
> As of now it doesn't do much besides allowing admins to shift mobs to a different center, but the long-term use of this is to be able to move the mobs up to give a better sense of depth perspective.

## Changelog
:cl: fighterslam, Kapu1178, Azarak, Gandalf2k15, Rohesie
add: You can adjust your current-tile location by "pixel shifting". Default key is C, check your binds.
code: Added base_pixel vars to atom/movable, replaced most uses of initial(pixel).
/:cl:

### CODE REVIEW

**Review by Kapu1178 — APPROVED** [review]

**Comment by Ms-Mee** [comment]
You are now under fear RP
Any actions that defy fear RP will result in termination.

**Comment by ZephyrTFA** [comment]
```
file ./code/game/atoms.dm is missing a trailing newline
file ./code/datums/keybinding/mob.dm is missing a trailing newline
file ./code/modules/pixelshifting/pixelshift.dm is missing a trailing newline
file ./code/modules/mob/living/living_defines.dm is missing a trailing newline
file ./code/modules/mob/mob_defines.dm is missing a trailing newline
```

**Comment by fighterslam** [comment]
thanks Zeph

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1531,3 +1535,21 @@
...
+
+///Setter for the "base_pixel_y" var to append behavior related to it's changing
+/atom/proc/set_base_pixel_y(new_value)
+	if(base_pixel_y == new_value)
+		return
+	. = base_pixel_y
+	base_pixel_y = new_value
+
+	pixel_y = pixel_y + base_pixel_y - .
+
```
Comment (ZephyrTFA) [inline code/game/atoms.dm]:
```suggestion

```

**Review by francinum — APPROVED** [review]
Looks good.

**Pull request merged**